codegen/smithy-dafny-codegen/src/test/java/software/amazon/polymorph/util/Tokenizer.java

```diff
@@ -33,6 +34,11 @@ public static record ParseToken(String text, int type) {}
     public static void tokenizeAndAssertEqual(String expected, String actual) {
         final List<ParseToken> actualTokens = tokenize(actual);
         final List<ParseToken> expectedTokens = tokenize(expected);
-        Assert.assertEquals(expectedTokens, actualTokens);
+        if (!Objects.equals(expectedTokens, actualTokens)) {
+            // If the tokens aren't equal, assert the original strings are equal
+            // knowing it is guaranteed to fail,
+            // just so that we get a much more readable diff in tooling.
+            Assert.assertEquals(expected, actual);
+        }
```

Comment:
Massive Praise: I really wish I had thought of this.
Would have made my life so much easier.